### ISSUE TITLE
Allow to create menus from stdin or configuration file

### ISSUE THREAD

**Issue by vegaelle** [nwg-piotr/sway-gtk-menu#49]
I’ve seen that [nwg-launchers](https://github.com/nwg-piotr/nwg-launchers) allow to create custom menu by passing those in the standard input. Additionnally, `sgtk-bar` can display any menu given the configuration file. Allowing this behaviour in `sgtk-menu` or `sgtk-dmenu` like `dmenu` does could allow to use this menus as a base for any kind of menu (for example, an emoji selector, or a MPD client as menu). Although for those examples, we should need to pass label/values and not simply names, as `nwgdmenu` does.

**Comment by nwg-piotr** [comment]
1. `sgkt-dmenu` should finally be given the ability to read from stdin. I've just been too lazy to add the missing code so far.
2. Creating `sgtk-menu` from a template was initially implemented in the sgtk-menu code.  Later I found it unnecessary and removed the feature, but I could be added back.
3. Building `sgtk-menu` on the basis of the stdin input exceeds what I've ever planned. Do you think someone is going to use such a feature?

**Comment by vegaelle** [comment]
1. would it work like `nwgdmenu`, with a list of values that would be displayed, or do you plan to accept labels/values? Would it be possible to pass icons and/or emojis as the labels (`sgtk-menu` does it, so it shouldn’t be too hard)?
3. i don’t know, i think having a flat menu would work (like rofi does, for example)

**Comment by nwg-piotr** [comment]
In case of `sgtk-dmenu`, the command and its label are the same:

```python
    for command in commands:  // those are just commands, even w/o the path
        item = Gtk.MenuItem.new_with_label(command)
        (...)
        item.connect('activate', launch, command)
```

...and this ain't gonna change. What you need, could only be implemented in `sgtk-menu`, on the basis of a template. It would still demand quite a lot of work.

**Comment by nwg-piotr** [comment]
Damn, I forgot that the python version still uses templates! :D  Check `-af` and `-n` arguments. This should display the items from a template only.

**Comment by vegaelle** [comment]
ooooh right, i already saw the `-af` which was interesting, but with the `-n` it takes far more sense! It will probably do, do you know if emojis could be passed as icons?

**Comment by nwg-piotr** [comment]
> do you know if emojis could be passed as icons?

You may use a registered icon name or a path to a graphics file.

I wonder if a template could be passed via stdin. This would allow to do some scripting. I'll take a closer look at it tonight.

**Comment by nwg-piotr** [comment]
Alright, it could look something like this. If a template in stdin given, the `-n | --no-menu` argument is being used automagically.

For instance:

> $ cat ~/path/to/template | sgtk-menu
> $ echo '[{"name":"Entry 1","exec":"swaymsg reload","icon":"reload"},{"name":"Entry 2","exec":"swaymsg reload"}]' | sgtk-menu

![pipemenu](https://user-images.githubusercontent.com/20579136/85477213-235c0880-b5ba-11ea-96d7-f010bb4873d6.png)

The "name" and "exec" fields are mandatory, "icon" is optional. Hope this is going to be enough for scripting. What do you think?

**Comment by nwg-piotr** [comment]
It seems to work quite well. Tested with a simple python script (sorry, I'm useless at bash):

*/home/piotr/PycharmProjects/test_pipemenu/main.py*

```python
# !/usr/bin/env python3

# find all downloaded pdf files, display menu to open them with Evince
from pathlib import Path

line = '['
for path in Path('/home/piotr/Pobrane').rglob('*.pdf'):
    line += '{'
    line += '"name": "%s", "exec": "evince \\"%s\\"", "icon": "evince"' % (path, path)
    line += '},'
# strip trailing comma
line = line[:-1]
line += ']'
print(line)

```

![test_pipemenu](https://user-images.githubusercontent.com/20579136/85487265-8b1b4f00-b5cc-11ea-8a80-06c7c8799341.png)

**Comment by nwg-piotr** [comment]
Released as v1.4.1. See [Wiki](https://github.com/nwg-piotr/sgtk-menu/wiki/Pipemenus).
